docs/DeveloperGuide.md

```diff
@@ -1,13 +1,413 @@
+`execute` to execute the feature.
+
+##### 3.4.1 and 3.5.1 XML Sequence Diagram
+
+The following sequence diagram shows the process of what happens when a user input is passed through the application,
+until it gets executed.
+
+**Note**: 
+1. Validator class is not a parent class, but we have grouped it under validators to prevent cluttering of 
+the diagrams. 
+2. Validator also returns different type of Command Classes depending on the validity of input, which has been omitted
+for the same reason as above.
+
+![CommandClass.drawio.png](diagrams/CommandClass.drawio.png)
+
+#### 3.6 Expense and Income Class
+The `Expense` and `Income` class inherits from the Transaction class.
+
+`Expense` class stores one expense record given by the user.
+
+`Income` class stores one income record given by the user.
+
+The methods for `Expense` and `Income` are not shown.
+
+
+![ExpenseAndIncomeClassDiagram.drawio.png](diagrams/ExpenseAndIncomeClassDiagram.drawio.png)
+
+### 4. Implementation
+
+#### 4.1 Add Expense Feature
+The Add Expense feature enables users to add budgets for different categories. This functionality is controlled by the
+AddExpenseCommand class, which is produced by the Parser class based on user input. The AddExpenseCommand class uses an
+AddExpenseValidator class to validate the provided description, amount, category, and date, and then create and add the
+given information to `AddExpenseCommand` if valid. Below is the relevance of these attributes:
+
+| Class Attribute | Variable Type | Relevance                            |
+|-----------------|---------------|--------------------------------------|
+| description     | String        | The short description of the expense |
+| amount          | double        | The expense amount to be added       |
+| category        | Category      | The category of expense to be added  |
+| date            | LocalDate     | The date of the expense              |
+
+The BudgetBuddy class then calls the `execute()` method of the `AddExpenseCommand` object which uses the following
+method in the `ExpenseManager` class to add the expense:
+
+| Method              | Return Type | Relevance                                 |
+|---------------------|-------------|-------------------------------------------|
+| addExpense(expense) | void        | Add new expense to the list of `expenses` |                                 
+
+A `RemainingBudgetManager` object will be created to find the budget remaining for the given month and category. 
+Finally, the acknowledgement message along with the budget remaining is displayed to the user using the `Ui` class 
+`displayToUser()` method.
+
+The following UML Sequence diagram shows how the Parser works to obtain the relevant inputs for the Add Expense Feature:
+The scenario of invalid input and sequence within `RemainingBudgetManager` are omitted to reduce complexity.
+![AddExpenseSequenceDiagram.drawio.png](diagrams/AddExpenseSequenceDiagram.drawio.png)
```

Comment:
Is alt the right frame to use? Maybe opt should be considered if there is no else path
![image](https://github.com/user-attachments/assets/484bec1e-323d-495d-8bac-0804abb86d4e)